README.md

```diff
@@ -1,2 +1,289 @@
 # ci-server-g26
-Assignment 2 for Fundamentals of Software Engineering at KTH, by Group 26.
+
+## TOC
+
+- [The CI Server](#the-ci-server)
+    - [The Personal Access Token](#the-personal-access-token---very-important!)
+- [HTTP Server](#http-server)
+- [Function Deep Dive](#function-deep-dive)
+    - [cloneReposiory()](#cloneRepository)
+    - [buildRepo()](#buildRepo)
+    - [sendResponse()](#sendResponse)
+- [Statement of Contributions](#statement-of-contributions)
+- [Essence Standard](#essence-standard)
+
+This repo is the codebase for a continuous integration server.   
+The server acts as an automated build status reporting tool that communicates 
+with Github on the functionality of commited code. The process is as follows:
+
+1. code is pushed to Github from a local repo
+2. Github registers the push and sends a json package to the server
+3. the server clones the repo, 
+4. builds the resultant code, 
+5. formats a response destined for Github based on the build result and
+6. sends the response (headers and json) back to Github
+
+The response back to Github contains the status of the build, which can either be 
+`error`, `faliure`, `pending` or `success`. This response is displayed 
+alongside the commit in Github.
```

Comment:
We only send FAILURE and SUCCESS actually

Reply:
yeah, but there I was more explaining what a webhoook can actually do. I guess the language is a bit unclear

Reply:
ahh yeah I get it now, no need to change then